u2f-ref-code/java/src/com/google/u2f/server/messages/RegisteredKey.java

```diff
@@ -117,9 +120,9 @@ public JsonObject getJson() {
     result.addProperty("version", version);
     result.addProperty("keyHandle", keyHandle);
     result.addProperty("sessionId", sessionId);
-    JsonArray transportsJson = getTransportsAsJson();
-    if (transportsJson != null) {
-      result.add("transports", transportsJson);
+    String transportsString = getTransportsAsString();
+    if (transportsString != null) {
```

Comment:
That's actually correct: if the transports are present (non-null) but empty, we want to send the empty string.

On a different note, the current source depends on Java7, but String.join only appears in Java8. I'll rewrite this line the uglier way.